### DISCUSSION TITLE
ioBroker: Sonoff ZbBridge via TCP

### DISCUSSION THREAD

**Discussion by Mike-CPT** [Koenkk/zigbee2mqtt#18479] [Devices]
Hi,
I am trying to configure the zigbee2mqtt adapter in ioBroker. The adapter installation finished without errors.

ioBroker is installed on a Rasp Pi running on 192.168.166.200.
I am using the ioBroker built-in mqtt broker on 192.168.166.200:1883 for all my devices.

I am using a Sonoff ZbBridge and followed "https://templates.blakadder.com/sonoff_ZBBridge.html" to enable tcp requests (via 192.168.166.228:888). 
The Zigbee adapter is able to talk to the Bridge by specifying tcp://192.168.166.228:888, instead of the serial port (e.g. /dev/ttyACM0).

May I ask if anyone can help me setting up the zigbee2mqtt adapter correctly to use it in my environment, without a Zigbee USB dongle connected to the Pi.

Thanks,
Mike

**Comment by kirovilya** [comment]
add adpater name "ezsp"
```
serial:
  adapter: ezsp
```

**Reply by Mike-CPT** [reply]
Thanks a lot, but where do I add/specify the new adapter? And what do I need to select/specify in the adapter GUI in ioBroker?

**Reply by kirovilya** [reply]
in iobroker.zigbee there is a choice of adapter type

**Reply by Mike-CPT** [reply]
Yes, I set that adapter type in iobroker.zigbee. As mentioned, iobroker.zigbee connects to the ZbBridge, but zigbee2mqtt doesn't. 
But above you specified a config file, where do I find that one? Unfortunately, I wasn't able to find proper configuration guidelines anywhere for zigbee2mqtt on ioBroker using ezsp.....and the settings GUI is a bit

**Reply by kirovilya** [reply]
https://www.zigbee2mqtt.io/guide/configuration/adapter-settings.html#configuration-of-the-zigbee-adapter

the configuration file can be found in the `data `folder where z2m is installed

**Reply by Mike-CPT** [reply]
The adapter is installed here: /opt/iobroker/node_modules/iobroker.zigbee2mqtt.
Please keep in mind, I am using ioBroker on Pi, not Home Assistant. 
But there is no "data" folder. However, does this mean, I need to configure the adapter via a config file and not via the ioBroker GUI (which would not be normal for ioBroker at all).

**Reply by kirovilya** [reply]
:) ioBroker.zigbee2mqtt this adapter implies that you already have z2m installed somewhere and that's where you need to make set up configuration.
in the adapter itself, the settings for connecting to the installed z2m are indicated.

**Reply by Mike-CPT** [reply]
Hi Ilya,
many thanks for your help. 
But I am not able to find that location. May I ask you for a path for ioBroker on Pi? 
Usually, adapters are very easy to configure in ioBroker, no config file modifications are needed. I am a bit surprised that it is such a mission for zigbee2mqtt...
Am I not able to configure everything required via the GUI?
